src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -533,7 +517,12 @@ case class IndexLogEntry(
     case _ => false
   }
 
-  def numBuckets: Int = derivedDataset.properties.numBuckets
+  def numBuckets: Int = {
```

Comment:
If there is a case where this property does not make sense let's have it `Optional` instead using `-1`.